lib/spack/spack/config.py

```diff
@@ -772,13 +848,40 @@ def _mark_internal(data, name):
     return d
 
 
-def _merge_yaml(dest, source):
+def type_of(path):
```

Comment:
I would call it:
```suggestion
def any_valid_type(path):
```
since there can be more than one valid choice (e.g. places where we accept both a string and a list of strings).

Reply:
Changed.